packages/beacon-node/src/node/nodejs.ts

```diff
@@ -159,6 +160,47 @@ export class BeaconNode {
       createLibp2pMetrics(libp2p, metrics.register);
     }
 
+    if (opts.chain.forwardWSCheckpoint !== undefined) {
+      const {epoch, root} = opts.chain.forwardWSCheckpoint;
+      // lets just check if we have the root in the finalized block archive, if not found then we should be syncing it
+      const block = (await db.blockArchive.getByRoot(root))?.message;
```

Comment:
Could you modularize this as:

```ts
if (opts.chain.forwardWSCheckpoint !== undefined) {
  const block = (await db.blockArchive.getByRoot(root))?.message;
  performForwardWSCheckpointCheck(...args)
}
```
And then unit test `performForwardWSCheckpointCheck`? Since this logic is really critical could you explain each case and play around with the if statements to see if the can be un-nested?